lib/msf/core/db_manager/route.rb

```diff
@@ -17,10 +23,14 @@ def report_session_route(session, route)
   }
   end
 
-  def report_session_route_remove(session, route)
+  def report_session_route_remove(opts)
     return if not active
+    session = opts[:session]
+    route = opts[:route]
     if session.respond_to? :db_record
       s = session.db_record
+    elsif session.is_a? Integer
+      s = Mdm::Session.find(session)
```

Comment:
There's explicit connecting pooling later in this method, is connecting pooling required here too?

Reply:
ohh uhh potentially yea
I thhough about using this instead aswell https://github.com/rapid7/metasploit-framework/blob/f8ee7eea9fd4ef820f46f9951e7464001e1a8b35/lib/msf/core/db_manager/session.rb#L7
but I wasn't sure if that's the sort of thign we wanted to be doing

Reply:
`Mdm::Session.find(session)` is a simply query on one ID. The pooling might be in place due to the large number of responses one might get back from a query against a subnet and mask (`find_by_subnet_and_netmask`).